File: core/src/main/java/at/shockbytes/corey/common/core/util/RunUtils.java
package at.shockbytes.corey.common.core.util;

import org.joda.time.Period;
import org.joda.time.PeriodType;
import org.joda.time.format.PeriodFormatter;
import org.joda.time.format.PeriodFormatterBuilder;


/**
 * @author Martin Macheiner
 *         Date: 10.09.2017.
 */

public class RunUtils {

    public static String calculatePace(long timeInMs, double distance) {

        if (distance <= 0) {
            return "-:--";
        }
        long kmMillis = (long) (timeInMs / distance);
        return formatPaceMillisToString(kmMillis);
    }

    public static int calculateCaloriesBurned(double distance, double weightOfRunner) {
        double burned = distance * weightOfRunner * 1.036;
        return (int) Math.floor(burned);
    }

    private static String formatPaceMillisToString(long kmMillis) {

        PeriodFormatter formatter = new PeriodFormatterBuilder()
                .minimumPrintedDigits(2)
                .appendMinutes()
                .appendSeparator(":")
                .appendSeconds()
                .toFormatter();

        PeriodType minutesSeconds = PeriodType.time()
                .withMillisRemoved()
                .withHoursRemoved();

        Period kmPeriod = new Period(kmMillis, minutesSeconds);
        return formatter.print(kmPeriod);
    }

}

File: core/src/main/java/at/shockbytes/corey/common/core/util/RunUtils.kt
package at.shockbytes.corey.common.core.util

import org.joda.time.Period
import org.joda.time.PeriodType
import org.joda.time.format.PeriodFormatterBuilder

/**
 * Author:  Martin Macheiner
 * Date:    10.09.2017
 */
object RunUtils {

    fun calculatePace(timeInMs: Long, distance: Double): String {

        if (distance <= 0) {
            return "-:--"
        }
        val kmMillis = (timeInMs / distance).toLong()
        return formatPaceMillisToString(kmMillis)
    }

    fun calculateCaloriesBurned(distance: Double, weightOfRunner: Double): Int {
        val burned = distance * weightOfRunner * 1.036
        return Math.floor(burned).toInt()
    }

    private fun formatPaceMillisToString(kmMillis: Long): String {

        val formatter = PeriodFormatterBuilder()
                .minimumPrintedDigits(2)
                .appendMinutes()
                .appendSeparator(":")
                .appendSeconds()
                .toFormatter()

        val minutesSeconds = PeriodType.time()
                .withMillisRemoved()
                .withHoursRemoved()

        val kmPeriod = Period(kmMillis, minutesSeconds)
        return formatter.print(kmPeriod)
    }
}
